tests/functional/conftest.py

```diff
@@ -0,0 +1,268 @@
+            endpoint = controller_data["controllers"][current_controller][
+                "api-endpoints"
+            ][0]
+            assert current_controller is not None
+            assert cacert is not None
+            assert endpoint is not None
+        except yaml.YAMLError as e:
+            logging.error(e)
+
+    with open(juju_account_file) as account_file:
+        try:
+            account_data = yaml.safe_load(account_file)
+            user = account_data["controllers"][current_controller]["user"]
+            password = account_data["controllers"][current_controller]["password"]
+        except yaml.YAMLError as e:
+            logging.error(e)
+
+    return user, password, cacert, endpoint
+
+
+def generate_config_data(user, password, cacert, endpoint):
+    """Generate config data with the parsed juju info."""
+    return {
+        "customer": {"name": "example_customer", "cloud_name": "example_cloud"},
+        "juju": {
+            "controller_endpoint": endpoint,
+            "controller_cacert": cacert,
+            "username": user,
+            "password": password,
+        },
+        "exporter": {
+            "port": 5000,
+            "collect_interval": int(COLLECTION_INTERVAL / 60),
+        },
+    }
+
+
+def configure_snap(version):
+    """Configure the snap."""
+    user, password, cacert, endpoint = get_juju_data()
+    config_data = generate_config_data(user, password, cacert, endpoint)
+
+    temp_config_file = os.path.join(TMP_DIR, "prometheus_juju_exporter_config.yaml")
+    with open(temp_config_file, "w") as config_file:
+        try:
+            yaml.dump(config_data, config_file)
+        except yaml.YAMLError as e:
+            logging.error(e)
+
+    snap_config_file = os.path.join(SNAP_CONFIG_DIR, version, "config.yaml")
+    assert (
+        check_call(f"sudo mv {temp_config_file} {snap_config_file}".split())
+        == 0  # noqa
+    )
+
+    assert os.path.isfile(snap_config_file)
+
+
+def get_snap_version():
+    """Get the version of the installed snap."""
+    try:
+        output = (
+            check_output(f"snap info {SNAP_NAME}".split()).decode().strip().splitlines()
+        )
+        package_info = [i for i in output if "installed:" in i]
+        version = package_info[0].split("(")[1].split(")")[0]
+    except Exception as e:
+        logging.error("Error getting snap version: %s", e)
+
+    return version
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_snap():
+    """Install the package to the system and cleanup afterwards.
+
+    An environment variable TEST_SNAP is needed to install the snap.
+    """
+    test_snap = os.environ.get("TEST_SNAP", None)
+    if test_snap:
+        logging.info("Installing %s snap package...", test_snap)
+        assert os.path.isfile(test_snap)
+        assert (
+            check_call(f"sudo snap install --dangerous {test_snap}".split())
+            == 0  # noqa
+        )
+
+        version = get_snap_version()
```

Comment:
I think that it'd be easier to define `SNAP_CONFIG_DIR` as 
```
SNAP_CONFIG_DIR = "/var/snap/prometheus-juju-exporter/current/"
```
rather than having the the `get_snap_version` and parsing the explicit version every time. From what I see the `get_snap_version` is not used anywhere else, right?